configs/module_merge.go

```diff
@@ -42,6 +42,10 @@ func (v *Variable) merge(ov *Variable) hcl.Diagnostics {
 		v.Description = ov.Description
 		v.DescriptionSet = ov.DescriptionSet
 	}
+	if ov.SensitiveSet {
```

Comment:
🤔 A random thought, triggered by PR but not requiring action (besides thinking, which in 2020 feels like a 4-hour workout):

I wonder if folks will ask for `sensitive=true` to take precedent during a merge (which contradicts our documented behavior). Granted if someone was very worried about this, my suggestion would be that using terraform enterprise+sentinel is their best bet for governance, and not solely `sensitive`.

Here's an example of what I'm thinking about:
Given this main.tf, I get the expected error because I'm pushing a sensitive var to an output:

main.tf:
```hcl
variable  "map" {
  type = map(string)
  default = {}
  sensitive = true
}
```

```
Error: Output refers to sensitive values

  on main.tf line 14:
  14: output "map" {

Expressions used in outputs can only refer to sensitive values if the
sensitive attribute is true.
```

But I can override that in main_override.tf:

```hcl
variable "map" {
  default = {
    "hi" = "there"
  }
  sensitive = false
}
```

And apply:
```
Apply complete! Resources: 0 added, 0 changed, 0 destroyed.

Outputs:

map = tomap({
  "hi" = "there"
})
```

Again, sorry to go off on a random tangent - this PR is perfect and this experiment is fantastic, but I wanted to bring up a security(ish) question I can imagine some organizations having.

Reply:
I think that's a great question! I'm going to write this down in the project's "open questions" because as you've pointed out, changing this would depart from our current documented behavior for how overrides work. It might well be worth it for this feature (to have specialized behavior for it) so thanks for bringing it up.